src/reconcile/rec_visibility.c

```diff
@@ -461,18 +461,24 @@ __rec_validate_upd_chain(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *s
      * reconciliations ondisk value that we will be comparing against.
      */
     if (vpack != NULL && !vpack->tw.prepare) {
-        WT_ASSERT_ALWAYS(session,
-          prev_upd->prepare_state == WT_PREPARE_INPROGRESS ||
-            prev_upd->start_ts == prev_upd->durable_ts ||
-            prev_upd->durable_ts >= vpack->tw.durable_start_ts,
-          "Durable timestamps cannot be out of order for prepared updates");
-        WT_ASSERT_ALWAYS(session,
-          prev_upd->prepare_state == WT_PREPARE_INPROGRESS ||
-            prev_upd->start_ts == prev_upd->durable_ts || !WT_TIME_WINDOW_HAS_STOP(&vpack->tw) ||
-            prev_upd->durable_ts >= vpack->tw.durable_stop_ts,
-          "Durable timestamps cannot be out of order for prepared updates");
-        if (prev_upd->start_ts < vpack->tw.start_ts ||
-          (WT_TIME_WINDOW_HAS_STOP(&vpack->tw) && prev_upd->start_ts < vpack->tw.stop_ts)) {
+        if (prev_upd->prepare_state == WT_PREPARE_INPROGRESS ||
```

Comment:
I thought we agree it do

if (WT_TIME_WINDOW_HAS_STOP(&vpack->tw))
     WT_ASSERT_ALWAYS(session,
               prev_upd->prepare_state == WT_PREPARE_INPROGRESS || prev_upd->start_ts == prev_upd->durable_ts ||
                prev_upd->durable_ts >= vpack->tw.durable_stop_ts,
              "Stop: Durable timestamps cannot be out of order for prepared updates");
   else
             WT_ASSERT_ALWAYS(session, prev_upd->prepare_state == WT_PREPARE_INPROGRESS || prev_upd->start_ts == prev_upd->durable_ts || prev_upd->durable_ts >= vpack->tw.durable_start_ts,
              "Start: Durable timestamps cannot be out of order for prepared updates");

Reply:
Wouldn't this be best?
```
        if (prev_upd->prepare_state != WT_PREPARE_INPROGRESS && prev_upd->start_ts != prev_upd->durable_ts) {
            WT_ASSERT_ALWAYS(session, prev_upd->durable_ts >= vpack->tw.durable_start_ts && (!WT_TIME_WINDOW_HAS_STOP(&vpack->tw) ||  prev_upd->durable_ts >= vpack->tw.durable_stop_ts));

        }
```

Reply:
I prefer my version because I think the code is clearer in that way. We compare the stop time pair if on the disk image we have one. Otherwise, we compare with the start time pair.

Reply:
Okay fair enough.

Reply:
Done